### PR TITLE
Replaced ActionWorldParamsUpdateVisitor with updateWorldParams

### LINKED ISSUE #1808: Replace ActionWorldParamsUpdateVisitor with an override method in each action
### Description of the task

Replace ActionWorldParamsUpdateVisitor with Action::updateWorldParams(World world), which each Action will override. This will simplify the code, because the complexity involved with ActionWorldParamsUpdateVisitor doesn't add much. If nothing else is using the ActionVisitor, it should be safe to remove.

### Acceptance criteria

See https://github.com/UBC-Thunderbots/Software/issues/1710 and https://github.com/UBC-Thunderbots/Software/pull/1802 for inspiration
- [ ] Replace ActionWorldParamsUpdateVisitor with Action::updateWorldParams
- [ ] Delete Action Visitor, provided nothing is using it

### Blocked By
N/A

### PR DESCRIPTION
**Pull request by xyztiger**
### Description

Replaces the action_visitor_update_world_params class with Action::updateWorldParams(World world)

### Testing Done

Updated all pre-existing tests with new function

### Resolved Issues

resolves #1808 

### Review Checklist

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
overall lgtm

you also need to go into `stp.cpp` and change the `action->accept(action_world_params_update_visitor)` to `action->updateWorldParams(world)` and delete the visitor

File: src/software/ai/hl/stp/action/action.h [hunk 1]
```diff
@@ -75,6 +76,13 @@ class Action
      */
     virtual void accept(MutableActionVisitor &visitor) = 0;
```
Comment (jonathanlew) [inline src/software/ai/hl/stp/action/action.h]: The MutableActionVisitor can be completely deleted as well

**Review by jonathanlew — CHANGES_REQUESTED** [review]

**Review by jonathanlew — DISMISSED** [review]
LGTM!

File: src/software/ai/hl/stp/action/stop_action.h [hunk 2]
```diff
@@ -27,15 +27,13 @@ class StopAction : public Action
...
 
     /**
-     * Updates the params that cannot be derived from the world for this action
+     * Updates the params that can be derived from the world for this action
      *
-     * @param robot the robot that should stop
-     * @param coast Whether or not to coast to a stop. If this is false, the robot will
-     * try actively brake to come to a stop
+     * @param ball The world to update from
      */
```
Comment (akhilveeraghanta) [inline src/software/ai/hl/stp/action/stop_action.h]: This comment is outdated, you can remove it

**Review by akhilveeraghanta — CHANGES_REQUESTED** [review]
Just one more nit!

File: src/software/ai/hl/stp/action/stop_action.h (hunk 2 above)
Comment (xyztiger) [reply src/software/ai/hl/stp/action/stop_action.h]: thanks, fixed :)

**Review by jonathanlew — APPROVED** [review]

**Review by akhilveeraghanta — APPROVED** [review]
nice :100:

**Pull request merged**